### PR TITLE
TAUR-1201: split up amqp.py into three files in new directory

### PR DESCRIPTION
**Pull request by tomsilver**
@vitaly-krugl Bringing this PR over from products repo.

### CODE REVIEW

File: nta.utils/nta/utils/amqp/constants.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+
+class AMQPErrorCodes(object):
+  """ AMQP Error Codes of interest; these occur in the method frame of
+  Channel.Close and Connection.Close methods
+  """
+
+  # Requested resource not found
+  NOT_FOUND = 404
+
```
Comment (vitaly-krugl) [inline nta.utils/nta/utils/amqp/constants.py]: Need to configure your editor and get rid of trailing spaces

File: nta.utils/nta/utils/amqp/messages.py [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+
+
+  def __repr__(self):
+    args = (
+      ("%s=%r" % (attr, getattr(self, attr)))
+      for attr in sorted(self.__slots__)
+      if not callable(getattr(self, attr)) and getattr(self, attr) is not None)
+
+    return "%s(%s)" % (self.__class__.__name__, ", ".join(args))
+
```
Comment (vitaly-krugl) [inline nta.utils/nta/utils/amqp/messages.py]: Need 3 lines between module-global declarations

**Comment by vitaly-krugl** [comment]
@tomsilver, just a couple of minor nits this time (whitespace-realted) in the pertinent code. However, I am concerned that this PR picked up several unrelated pipeline changes. Was this intended?

**Comment by tomsilver** [comment]
@vitaly-krugl @scottpurdy Vitaly gave me the LGTM - can you merge if it looks good?

**Comment by vitaly-krugl** [comment]
@tomsilver, if it passes all the tests in question in numenta-apps, go ahead and merge.

**Comment by tomsilver** [comment]
I don't think I have the permissions necessary to merge?

**Comment by vitaly-krugl** [comment]
@tomsilver, now that you mention it, I don't either. 

@scottpurdy @jcasner: neither Tom nor I seem to have write access here. Is that intended?

**Comment by scottpurdy** [comment]
I don't think it is intended. @jcasner ?

File: grok/grok/app/runtime/notification_service.py [hunk 3]
```diff
@@ -34,7 +35,11 @@
 from pytz import timezone
 import requests
 
-from nta.utils import amqp
+from nta.utils.amqp import connection as amqp_connection
```
Comment (scottpurdy) [inline grok/grok/app/runtime/notification_service.py]: I would strongly suggest you import `amqp` and then reference these as `amqp.connection` and similar. That way you still have the context and you don't have to rename the imports.

Comment (vitaly-krugl) [inline grok/grok/app/runtime/notification_service.py]: amqp.py would need to import connection and friends to make this possible.

**Comment by tomsilver** [comment]
@scottpurdy I changed all instances of amqp_ to amqp. as we discussed. Could you let me know how it looks and if it's ok to merge? Thanks!

File: nta.utils/nta/utils/amqp/__init__.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+from . import connection
+from . import constants
+from . import consumer
+from . import exceptions
+from . import messages
+from . import queue
+from . import synchronous_amqp_client
```
Comment (vitaly-krugl) [inline nta.utils/nta/utils/amqp/__init__.py]: Need trailing newline

**Comment by vitaly-krugl** [comment]
Thanks @tomsilver. I like this result better, too.

**Pull request merged**